apis/python/src/tiledb/vector_search/ingestion.py

```diff
@@ -2333,6 +2390,54 @@ def scale_resources(min_resource, max_resource, max_input_size, input_size):
                     **kwargs,
                 )
             else:
+                # If the distance metric is cosine, we need to create a node that everything depends on
+                # Which reads the vectors and normalizes them, then swaps source_uri for normalized_uri
+                # This is because the cosine distance metric requires normalized vectors
+                normalization_nodes = []
+                if distance_metric == vspy.DistanceMetric.COSINE:
+                    group = tiledb.Group(index_group_uri, "w")
+                    normalized_uri = create_array(
+                        group=group,
+                        size=size,
+                        dimensions=dimensions,
+                        vector_type=vector_type,
+                        array_name="normalized_vectors",
+                    )
+                    group.close()
+
+                    # Create normalization nodes
+                    for start in range(
+                        0, size, input_vectors_batch_size_during_sampling
+                    ):
+                        end = min(
+                            start + input_vectors_batch_size_during_sampling, size
+                        )
+
+                        normalization_nodes.append(
+                            submit(
+                                normalize_source_vectors,
+                                source_uri=source_uri,
+                                normalized_uri=normalized_uri,
+                                source_type=source_type,
+                                vector_type=vector_type,
+                                dimensions=dimensions,
+                                source_start_pos=start,
+                                source_end_pos=end,
+                                batch=input_vectors_per_work_item_during_sampling,
+                                config=config,
+                                verbose=verbose,
+                                trace_id=trace_id,
+                                name=f"normalize-vectors-{start}-{end}",
+                                resources=random_sample_resources,  # We can use similar resources as random sampling
```

Comment:
(Minor) I think the code is self-explanatory enough?
```suggestion
                                resources=random_sample_resources,
```